src/tmpfiles/tmpfiles.c

```diff
@@ -3067,12 +3101,14 @@ static int process_item(
         if (i->allow_failure)
                 r = 0;
 
-        q = FLAGS_SET(operation, OPERATION_REMOVE) ? remove_item(c, i) : 0;
-        p = FLAGS_SET(operation, OPERATION_CLEAN) ? clean_item(c, i) : 0;
+        q = FLAGS_SET(operation, OPERATION_REMOVE) || FLAGS_SET(operation, OPERATION_PURGE_OWNED) ? remove_item(c, i) : 0;
+        p = FLAGS_SET(operation, OPERATION_CLEAN) || FLAGS_SET(operation, OPERATION_PURGE_OWNED) ? clean_item(c, i) : 0;
```

Comment:
hmm, why this? it appears to me that cleaning a dir we are about to purge is a bit pointless?

Reply:
I think it would be nicer to just allow people to specify `--purge --remove --clean` if they want on the command line.

It's not obvious to me at all that there should be an implied superset here